overlord/ifacestate/handlers.go

```diff
@@ -487,8 +487,24 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		}
 	}
 
-	cref := interfaces.ConnRef{PlugRef: plugRef, SlotRef: slotRef}
-	if conn, ok := conns[cref.ID()]; ok && conn.Auto {
+	cref := interfaces.ConnRef{
+		PlugRef: interfaces.PlugRef{Snap: RemapSnapFromState(plugRef.Snap), Name: plugRef.Name},
+		SlotRef: interfaces.SlotRef{Snap: RemapSnapFromState(slotRef.Snap), Name: slotRef.Name}}
```

Comment:
I don't understand why we need this change,  the tasks should not have from-state values of them,  if we have snap d in the system the task should carry snapd (not core). If you look at daemon at most what we need is RemapSnapFromRequest somewhere.